### PR TITLE
[12.0][MIG][WIP] Migrate DNS Connector from 8.0 to 12.0

### PR DESCRIPTION
**Pull request by ruter-lyu**
Todo:

- [ ] Add readme

### CODE REVIEW

**Comment by elicoidal** [comment]
Please check travis @ruter-lyu

**Comment by elicoidal** [comment]
Please check travis @ruter-lyu 
I am not sure how you preserved the commit history but it looks like I miss some from https://github.com/OCA/infrastructure-dns/commits/8.0/connector_dns

see https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

besides you have: https://github.com/OCA/infrastructure-dns/pull/6

File: connector_dns/components/deleter.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class DNSAbstractDeleter(AbstractComponent):
+    _name = 'dns.abstract.deleter'
+    _inherit = 'base.deleter'
+    _usage = 'dns.deleter'
+
+    def run(self, binding):
+        self._before_delete(binding)
+        self.backend_adapter.delete(binding)
```
Comment (JoJoJoJoJoJoJo) [inline connector_dns/components/deleter.py:13]: unlink or delete?

File: connector_dns/models/dns_backend.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from odoo import models, fields, api
+
+
+class DNSBackend(models.Model):
+    _name = 'dns.backend'
+    _inherit = 'connector.backend'
+    _description = 'DNS Connector Backend'
+
+    @api.model
+    def _selection_version(self):
```
Comment (JoJoJoJoJoJoJo) [inline connector_dns/models/dns_backend.py]: Remove this function.

File: connector_dns/models/dns_backend.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        return [('1.0', '1.0')]
+
+    name = fields.Char('Name', required=True)
+    company_id = fields.Many2one(
+        'res.company',
+        default=lambda self: self.env.user.company_id,
+        string='Company')
+    is_default = fields.Boolean('Is Default Backend')
+    import_date = fields.Datetime('Import Date')
```
Comment (JoJoJoJoJoJoJo) [inline connector_dns/models/dns_backend.py]: Remove the following 2 fields.

File: connector_dns/models/dns_binding.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    def export_dns_records(self, backend_id, binding):
+        with backend_id.work_on(self._name) as work:
+            exporter = work.component(usage='dns.exporter')
+            return exporter.run(binding)
+
+    @job(default_channel='root')
+    @api.model
+    def delete_dns_records(self, backend_id, binding):
+        with backend_id.work_on(self._name) as work:
+            exporter = work.component(usage='dns.deleter')
```
Comment (JoJoJoJoJoJoJo) [inline connector_dns/models/dns_binding.py]: exporter --> deleter

**Review by JoJoJoJoJoJoJo — CHANGES_REQUESTED** [review]

**Review by JoJoJoJoJoJoJo — APPROVED** [review]
LGTM, traivs

**Comment by flotho** [comment]
any plan to restart this ?

**Pull request closed without merging**